R/vdiagram-render.R

```diff
@@ -54,6 +54,129 @@ renderVDiagram <- function(data){
 }
 
 createVdiagram <- function(data) {
-  data <- parseVDiagramData(data)
-  vplot(data)
+  styles <- getVDiagramStyle()
+  
+  vdiagramData <- parseVDiagramData(data)
+  
+  vplot <- gsplot() %>%
+    points(NA,NA, ylab=styles$labels$y, xlab=styles$labels$x) %>%
+    callouts(y=c(vdiagramData$minStage, vdiagramData$maxStage), styles$callouts) %>%
+    grid(styles$grid) %>%
+    axis(styles$axis);
+  
+  vplot <- addMeasurementsAndError(vplot, vdiagramData, styles)
+  vplot <- addRatingShifts(vplot, vdiagramData, styles)
+  
+  if (any(!is.na(vdiagramData$obsShift)) && any(!vdiagramData$histFlag)){
```

Comment:
It would be nice if the return type of `histFlag` was such that we didn't need this logic, and instead were able to just call the following lines no matter what.